integration/helpers/s3.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+#shellcheck disable=SC2034
+
+__bucket_base_path() {
+    head /dev/urandom | tr -dc A-Za-z0-9 | head -c 13
+    return 0
+}
+
+s3_bucket_name="a2-backup-restore-test"
+
+s3_bucket_base_path="$(__bucket_base_path)"
+
+# deployment-service currently requires the regional endpoint
+s3_endpoint="https://s3.us-west-2.amazonaws.com"
+
+do_deploy_s3_default() {
+    #shellcheck disable=SC2154
+    chef-automate deploy config.toml \
+        --product automate \
+        --product builder \
+        --hartifacts "$test_hartifacts_path" \
+        --override-origin "$HAB_ORIGIN" \
+        --manifest-dir "$test_manifest_path" \
+        --admin-password chefautomate \
+        --accept-terms-and-mlsa
+
+    log_info "applying dev license"
+    chef-automate license apply "$A2_LICENSE"
+}
+
+do_test_deploy_s3_default() {
+    #shellcheck disable=SC2154
+    #shellcheck source=integration/helpers/bldr_tests.sh
+    source "${source_dir}/helpers/bldr_tests.sh"
+
+    do_test_deploy_default
+    bldr_smoke_test
+}
+
+do_backup_s3_default() {
+  echo "Backing up to S3"
+  echo "  bucket: ${s3_bucket_name}/${s3_bucket_base_path}"
+  echo "  endpoint: ${s3_endpoint}"
+  do_backup_default
+}
+
+do_prepare_restore_s3_default() {
+  do_prepare_restore_default
+
+  # Remove the default backup directory
+  [[ -d /var/opt/chef-automate/backups ]] && rm -r /var/opt/chef-automate/backups
+
+  # make sure we can list the backups, and it contains
+  # our backup
+  #shellcheck disable=SC2154
+  chef-automate backup list \
+      --debug \
+      "s3://${s3_bucket_name}/${s3_bucket_base_path}" | awk -v s="${test_backup_id}" 'BEGIN{r=1}; $0~s{r=0} 1; END{exit(r)}'
+}
+
+do_restore_s3_default() {
+  #shellcheck disable=SC2154
+  chef-automate backup restore \
+      --debug \
+      --override-origin "$HAB_ORIGIN" \
+      "s3://${s3_bucket_name}/${s3_bucket_base_path}/${test_backup_id}"
+}
+
+do_test_restore_s3_default() {
+  do_test_restore_default || return 1
+  delete_backup_and_assert_idempotent
+  verify_packages
+
+  hab pkg install core/hab
+  artifacts_dir=$(mktemp -d --suffix="bldr_smoke")
+  # TODO(jaym): remove when we upgrade hab to a version
+  # that supports hab pkg download
```

Comment:
Is this comment still needed, or is there something about the `hab pkg download` command we're running on the next line that isn't what it's going to be in the next version?

Reply:
I have no idea, I literally extracted the code from the existing test into a helper.